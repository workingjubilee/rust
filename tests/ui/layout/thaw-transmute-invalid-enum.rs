//@ known-bug: rust-lang/rust#126966
mod assert {
    use std::mem::{Assume, TransmuteFrom};

    pub fn is_transmutable<Src, Dst>()
    where
        Dst: TransmuteFrom<Src>,
    {
    }
}

#[repr(u32)]
enum Ox00 {
    V = 0x00,
}

#[repr(C, packed(2))]
enum OxFF {
    V = 0xFF,
}

fn test() {
    union Superset {
        a: Ox00,
        b: OxFF,
    }

    assert::is_transmutable::<Superset, Subset>();
}
