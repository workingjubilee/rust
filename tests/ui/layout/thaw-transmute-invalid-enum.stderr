error[E0412]: cannot find type `Subset` in this scope
  --> $DIR/thaw-transmute-invalid-enum.rs:28:41
   |
LL |     assert::is_transmutable::<Superset, Subset>();
   |                                         ^^^^^^ not found in this scope
   |
help: you might be missing a type parameter
   |
LL | fn test<Subset>() {
   |        ++++++++

error[E0601]: `main` function not found in crate `thaw_transmute_invalid_enum`
  --> $DIR/thaw-transmute-invalid-enum.rs:29:2
   |
LL | }
   |  ^ consider adding a `main` function to `$DIR/thaw-transmute-invalid-enum.rs`

error[E0517]: attribute should be applied to a struct or union
  --> $DIR/thaw-transmute-invalid-enum.rs:17:11
   |
LL |   #[repr(C, packed(2))]
   |             ^^^^^^^^^
LL | / enum OxFF {
LL | |     V = 0xFF,
LL | | }
   | |_- not a struct or union

error[E0658]: use of unstable library feature 'transmutability'
  --> $DIR/thaw-transmute-invalid-enum.rs:3:20
   |
LL |     use std::mem::{Assume, TransmuteFrom};
   |                    ^^^^^^
   |
   = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
   = help: add `#![feature(transmutability)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature 'transmutability'
  --> $DIR/thaw-transmute-invalid-enum.rs:3:28
   |
LL |     use std::mem::{Assume, TransmuteFrom};
   |                            ^^^^^^^^^^^^^
   |
   = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
   = help: add `#![feature(transmutability)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature 'transmutability'
  --> $DIR/thaw-transmute-invalid-enum.rs:7:14
   |
LL |         Dst: TransmuteFrom<Src>,
   |              ^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #99571 <https://github.com/rust-lang/rust/issues/99571> for more information
   = help: add `#![feature(transmutability)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0740]: field must implement `Copy` or be wrapped in `ManuallyDrop<...>` to be used in a union
  --> $DIR/thaw-transmute-invalid-enum.rs:24:9
   |
LL |         a: Ox00,
   |         ^^^^^^^
   |
   = note: union fields must not have drop side-effects, which is currently enforced via either `Copy` or `ManuallyDrop<...>`
help: wrap the field type in `ManuallyDrop<...>`
   |
LL |         a: std::mem::ManuallyDrop<Ox00>,
   |            +++++++++++++++++++++++    +

error: aborting due to 7 previous errors

Some errors have detailed explanations: E0412, E0517, E0601, E0658, E0740.
For more information about an error, try `rustc --explain E0412`.
